### ISSUE TITLE
[Feature request] Score confidence parameter.

### ISSUE THREAD

**Issue by LucaCappelletti94** [scikit-optimize/scikit-optimize#771]
Hello, I was wondering if there could be a way to assign confidence values to the scores.

Suppose I have a probabilistic algorithm that after `n` iterations assures me to have found the optimum with 70% probability and after `m` 90%. I may want to iterate with the bayesian optimizer over the number of iterations, but the prior update within the bayesian optimizer would require to include the score confidence.

How could I achieve this?

(P.S. If it makes any sense.)

**Comment by chnzhangrui** [comment]
Hi,

I am not sure I fully get your point. Do you want to estimate n_iter before doing a Bayesian optimisation to cover a certain probability (say 90%)? If so, I have two suggestions:
1) The random search usually provides good probability coverage. (see the [paper](http://www.jmlr.org/papers/volume13/bergstra12a/bergstra12a.pdf))
2) The Gaussian process has the possibility to provide a noise term to indicate the confidence of the evaluated score. I have to check if this is implemented in `skopt` anyway.

**Comment by LucaCappelletti94** [comment]
Hello! 

Maybe I was not able to explain myself, sorry. Let me try again:

Basically I was wondering if I can provide to the gaussian process a score function whose values are obtained using probabilistic algorithms.

My reasoning goes like this: first I could run a probabilistic algorithm for `n` tasks so that I obtain with a given probability (say 90%) the algorithm score. Then, afterwards, I could run the deterministic version only on the space of parameters suggested by the previous iterations, thus speeding up (hopefully) the process.

Does this idea sound reasonable to you? Thanks!

**Comment by chnzhangrui** [comment]
Hi,

To reiterate your idea: you want to run n points in the search space first to make sure at least one of these points can provide you with the maximum of the objective function with 90% probability.

Then you do a grid search (that is what I understand of _deterministic_ _version_) and find the true maximum?

Why do you think this will speed up the process?

**Comment by LucaCappelletti94** [comment]
Hello,

No grid search involved: basically the process would use two black-box functions, an approximated probabilistic one that returns the optimal solution with 90% probability of the optimum and a deterministic one that requires much more time but returns the optimum.

I would start for the first few samples with the probabilistic one, then I would switch to the deterministic one when the weights are at least partially selected.

I would love therefore to be able to weight in some way the update of the gaussian process when the score returned has a given probability to be the optimum.

**Comment by chnzhangrui** [comment]
Hi,
This can be done manually by two steps for the moment. Step one with a larger search space and run a random search. Step two is to shrink the space to the best points found in the first step. I haven't found a way to do this automatically though. Please share your opinions if you have an idea.
